### PR TITLE
Spark role

### PR DESCRIPTION
**Pull request by opadron**
- Adds new roles for provisioning spark and the spark cluster dispatcher service for mesos.
- Refactors other roles to be more consistent and to use more sensible defaults.
- Includes other minor improvements.

### CODE REVIEW

**Comment by opadron** [comment]
As requested, pinging @zachmullen, @mgrauer

File: doc/hadoop.md [hunk 1]
```diff
@@ -12,16 +12,16 @@ roles.
...
 |hdfs_net_interface         |eth0       |interface on which to bind                 |
-|hdfs_data_root             |/data/hdfs |root directory for the HDFS files          |
 |hdfs_block_replication     |3          |replication factor                         |
 |hdfs_block_size            |268435456  |block size in bytes                        |
 |hdfs_io_buffer_size        |131072     |IO buffer size in bytes                    |
 |state                      |started    |state of the service                       |
 |format                     |false      |whether to (re)format the filesystem       |
-|hadoop_version             |2.7.1      |version of hadoop to deploy                |
-|hadoop_install_root        |/opt/hadoop|root directory to install hadoop under     |
-|hdfs_namenode_ansible_group|(required) |ansible group name for the HDFS name nodes |
```
Comment (cpatrick) [inline doc/hadoop.md:24]: Can we alphabetize this table?

File: doc/mesos.md [hunk 2]
```diff
@@ -14,8 +14,8 @@ This section applies to the `mesos-master` and `mesos-slave` roles.
 |mesos_group                            |mesos             |group to run the mesos services as                                                              |
 |mesos_crypt_pass                       |(generated)       |hash of the password to use for the user                                                        |
 |mesos_version                          |0.21.0            |version of mesos to deploy                                                                      |
-|mesos_install_root                     |/opt/mesos        |root directory to install mesos under                                                           |
-|mesos_data_root                        |/data/mesos       |root directory for the mesos data files                                                         |
+|mesos_install_root                     |(generated)       |root directory to install mesos under                                                           |
+|mesos_data_root                        |(generated)       |root directory for the mesos data files                                                         |
```
Comment (cpatrick) [inline doc/mesos.md]: Again, alphabetize this table.

File: doc/spark.md [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+|hdfs_namenode_ansible_group          |(required)    |ansible group name for the hdfs name nodes                                   |
+|zookeeper_ansible_group              |(required)    |ansible group name for the zookeeper nodes                                   |
+|mesos_master_ansible_group           |(required)    |ansible group name for the mesos master nodes                                |
+|spark_net_interface                  |eth0          |interface on which to bind                                                   |
+|spark_driver_cores                   |(sparkdefault)|number of cores to use for the driver process                                |
+|spark_driver_memory                  |(sparkdefault)|amount of memory to use for the driver process                               |
+|spark_executor_memory                |(sparkdefault)|amount of memory to use per executor process                                 |
+|spark_local_dir                      |(sparkdefault)|directory to use for "scratch" space in spark                                |
```
Comment (cpatrick) [inline doc/spark.md:23]: alphabetize table.

File: doc/zookeeper.md [hunk 4]
```diff
@@ -11,8 +11,8 @@ Provisions and manages a zookeeper node
 |zookeeper_group                 |zookeeper      |group to run the zookeeper service as                                        |
 |zookeeper_crypt_pass            |(generated)    |hash of the password to use for the user                                     |
 |zookeeper_version               |3.4.6          |version of zookeeper to deploy                                               |
-|zookeeper_install_root          |/opt/zookeeper |root directory to install zookeeper under                                    |
-|zookeeper_data_root             |/data/zookeeper|root directory for the zookeeper data files                                  |
+|zookeeper_install_root          |(generated)    |root directory to install zookeeper under                                    |
+|zookeeper_data_root             |(generated)    |root directory for the zookeeper data files                                  |
```
Comment (cpatrick) [inline doc/zookeeper.md:17]: image

**Comment by cpatrick** [comment]
@opadron This is very nice. I love the consistency between the roles.

@kotfic If when/you do get AWS credentials, it would be cool to work with @opadron to try this out. Once you've done that, this LGTM.

**Comment by kotfic** [comment]
@cpatrick might not want to wait on that,  AWS credentials are on hold..  I would love to coordinate with @opadron to get this working with the vagrant PR though

**Comment by cpatrick** [comment]
@kotfic That's a shame :crying_cat_face:. @opadron, @kotfic, the vagrant PR sounds like a good approach to reproduce this work on a non-AWS environment. Let's give that a shot.

File: filter_plugins/gobig.py [hunk 5]
```diff
@@ -0,0 +1,19 @@
+
+def zk_url(list, hostvars, net_interface, client_port):
```
Comment (mgrauer) [inline filter_plugins/gobig.py]: very cool.  it all makes sense now!

Comment (zachmullen) [inline filter_plugins/gobig.py]: Can we change `list` to a different name so it doesn't shadow the builtin?

**Comment by opadron** [comment]
I've tested this PR, #7, #11, and #12 in a local integration branch and can confirm that things are working, minus a separate issue that seems to be affecting several roles.

**Pull request merged**